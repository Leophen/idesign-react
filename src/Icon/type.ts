interface IconProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 组件自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 图标名称
   */
  name?: string;
  /**
   * 图标尺寸
   * @default 16
   */
  size?: number | string;
  /**
   * 图标颜色
   */
  color?: string;
  /**
   * 是否禁用图标
   */
  disabled?: boolean;
  /**
   * 点击图标触发事件
   */
  onClick?: any;
}
