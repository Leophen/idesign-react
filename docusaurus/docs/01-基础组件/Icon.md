---
title: Icon 按钮
---

import IconDisplay from '/src/components/IconDisplay/index.tsx'

iDesign 提供了一套常用的图标集合，可直接使用。

## 基本用法

使用 `name` 属性选择展示的图标。

```tsx live
<div className="demo-block-row">
  <Icon name="ModeLight" />
  <Icon name="ModeDark" />
</div>
```

## 不同尺寸

提供 `size` 属性自定义图标大小。

```tsx live
<div className="demo-block-row">
  <Icon name="TipCheck" size={16} />
  <Icon name="TipCheck" size={24} />
  <Icon name="TipCheck" size={32} />
</div>
```

## 不同颜色

提供 `color` 属性自定义图标大小。

```tsx live
<div className="demo-block-row">
  <Icon name="TipClose" size={24} color="#5e62ea" />
  <Icon name="TipClose" size={24} color="#65b687" />
  <Icon name="TipClose" size={24} color="#f1ac51" />
  <Icon name="TipClose" size={24} color="#ec7491" />
</div>
```

## 全部图标

<IconDisplay />

## API

| 属性      | 说明             | 类型                         | 默认值 |
| --------- | ---------------- | ---------------------------- | ------ |
| className | 类名             | `string`                     | `--`   |
| style     | 组件自定义样式   | `CSSProperties`              | `--`   |
| name      | 图标名称         | `string`                     | `--`   |
| size      | 图标尺寸         | `string` \| `number`         | `16`   |
| color     | 图标颜色         | `string`                     | `--`   |
| disabled  | 是否禁用图标     | `boolean`                    | `--`   |
| onClick   | 点击图标触发事件 | `MouseEventHandler<Element>` | `--`   |
