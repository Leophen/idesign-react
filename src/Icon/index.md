---
nav:
  title: 组件
  path: /components
group:
  title: 基础组件
  order: 1
order: 2
---

# Icon 图标

iDesign 提供了一套常用的图标集合，可直接使用。

## 基本用法

使用 `name` 属性选择展示的图标。

```tsx
import React from 'react';
import { Icon } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Icon name="ModeLight" />
      <Icon name="ModeDark" />
    </div>
  );
};

export default App;
```

## 不同尺寸

提供 `size` 属性自定义图标大小。

```tsx
import React from 'react';
import { Icon } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Icon name="TipCheck" size={24} />
      <Icon name="TipCheck" size={32} />
      <Icon name="TipCheck" size={48} />
    </div>
  );
};

export default App;
```

## 不同颜色

提供 `color` 属性自定义图标大小。

```tsx
import React from 'react';
import { Icon } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Icon name="TipClose" size={32} color="#8A88FF" />
      <Icon name="TipClose" size={32} color="rgb(58, 54, 255)" />
      <Icon name="TipClose" size={32} color="hsl(211, 100%, 61%)" />
      <Icon name="TipClose" size={32} color="#19E6F3" />
    </div>
  );
};

export default App;
```

## 全部图标

```tsx
/**
 * inline: true
 */

import React, { useState, useEffect } from 'react';
import { Icon } from 'idesign-react';
import _ from 'lodash';
import axios from 'axios';
import classNames from 'classnames';

const App = () => {
  const [iconArr, setIconArr] = useState([]);
  const api =
    'https://at.alicdn.com/t/font_3161433_glke53nnqws.json?spm=a313x.7781069.1998910419.80&file=font_3161433_glke53nnqws.json';

  useEffect(() => {
    let isUnmounted = false;
    const abortController = new window.AbortController();
    axios
      .get(api)
      .then((res) => setIconArr(res.data.glyphs))
      .finally(() => {});
    return () => {
      isUnmounted = true;
      abortController.abort;
    };
  }, []);

  const sortedIconArr = _.sortBy(iconArr, (item) => item.name);

  const [copyed, setCopyed] = useState(false);
  const [curName, setCurName] = useState('');
  const resetCopyTip = () => {
    setCopyed(false);
    setCurName('');
    clearTimeout(resetCopyTip);
  };
  const handleCopyName = (value) => {
    navigator.clipboard.writeText(`<Icon name="${value}" />`);
    setCopyed(true);
    setCurName(value);
    setTimeout(resetCopyTip, 500);
  };

  return (
    <ul className="i-design-demo-icon-list">
      {sortedIconArr.map((item) => {
        return (
          <li
            className={classNames(
              'i-design-demo-icon-item',
              curName === item.name && copyed && '-copyed',
            )}
            key={item.icon_id}
            onClick={() => handleCopyName(item.name)}
          >
            <Icon name={item.name} size={32} />
            <div>{item.name}</div>
          </li>
        );
      })}
    </ul>
  );
};

export default App;
```

<API />
