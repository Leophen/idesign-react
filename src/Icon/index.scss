@font-face {
  font-family: 'idesignicon';
  src: url('');
  src: url('?#iefix') format('embedded-opentype'),
    url('//at.alicdn.com/t/font_3161433_glke53nnqws.woff2?t=1654516269726') format('woff2'),
    url('//at.alicdn.com/t/font_3161433_glke53nnqws.woff?t=1654516269726') format('woff'),
    url('//at.alicdn.com/t/font_3161433_glke53nnqws.ttf?t=1654516269726') format('truetype'),
    url('#iconfont') format('svg');
}

.i-icon {
  color: var(--i-font-3);
  font-size: 16px;
  font-family: 'idesignicon' !important;
  font-style: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.icon-StarFill:before {
  content: "\e690";
}

.icon-Dropper:before {
  content: "\e6bf";
}

.icon-Clock:before {
  content: "\e6be";
}

.icon-ViewHide:before {
  content: "\e6bc";
}

.icon-View:before {
  content: "\e6bd";
}

.icon-TagPrice:before {
  content: "\e6ba";
}

.icon-TagCollection:before {
  content: "\e6bb";
}

.icon-AddFile:before {
  content: "\e6b9";
}

.icon-CheckBold:before {
  content: "\e653";
}

.icon-SearchSub:before {
  content: "\e654";
}

.icon-UserFilled:before {
  content: "\e655";
}

.icon-ChatSquare:before {
  content: "\e656";
}

.icon-ArrowLineTopLeft:before {
  content: "\e657";
}

.icon-ShoppingCart:before {
  content: "\e658";
}

.icon-RefreshLeft:before {
  content: "\e659";
}

.icon-Filter:before {
  content: "\e65b";
}

.icon-Share:before {
  content: "\e65c";
}

.icon-SearchPlus:before {
  content: "\e65d";
}

.icon-Flag:before {
  content: "\e65e";
}

.icon-LocationFill:before {
  content: "\e65f";
}

.icon-DeleteFill:before {
  content: "\e660";
}

.icon-Histogram:before {
  content: "\e661";
}

.icon-LinkDouble:before {
  content: "\e662";
}

.icon-More:before {
  content: "\e663";
}

.icon-ArrowSortBottom:before {
  content: "\e664";
}

.icon-TipPlus:before {
  content: "\e665";
}

.icon-ChatLineSquare:before {
  content: "\e666";
}

.icon-Female:before {
  content: "\e667";
}

.icon-ChatDotSquare:before {
  content: "\e668";
}

.icon-ArrowSort:before {
  content: "\e669";
}

.icon-Search:before {
  content: "\e66a";
}

.icon-Copy:before {
  content: "\e66b";
}

.icon-ArrowDoubleRight:before {
  content: "\e66c";
}

.icon-TipSubFill:before {
  content: "\e66d";
}

.icon-TheSub:before {
  content: "\e66e";
}

.icon-Loading:before {
  content: "\e66f";
}

.icon-PictureRounded:before {
  content: "\e670";
}

.icon-Edit:before {
  content: "\e671";
}

.icon-TipQuestion:before {
  content: "\e672";
}

.icon-TipPlusFill:before {
  content: "\e673";
}

.icon-ShoppingCartFull:before {
  content: "\e674";
}

.icon-TipSub:before {
  content: "\e675";
}

.icon-TipInfo:before {
  content: "\e676";
}

.icon-ArrowCaretLeft:before {
  content: "\e677";
}

.icon-Star:before {
  content: "\e678";
}

.icon-Link:before {
  content: "\e679";
}

.icon-ChatLineRound:before {
  content: "\e67a";
}

.icon-TipWarningFill:before {
  content: "\e67b";
}

.icon-TipCheckFill:before {
  content: "\e67c";
}

.icon-Delete:before {
  content: "\e67d";
}

.icon-Bell:before {
  content: "\e67e";
}

.icon-LocationArrow:before {
  content: "\e67f";
}

.icon-ArrowLineUpload:before {
  content: "\e680";
}

.icon-ArrowCaretRight:before {
  content: "\e681";
}

.icon-ArrowCaretSort:before {
  content: "\e682";
}

.icon-SettingFill:before {
  content: "\e683";
}

.icon-User:before {
  content: "\e684";
}

.icon-Location:before {
  content: "\e685";
}

.icon-MoreFilled:before {
  content: "\e686";
}

.icon-ArrowLineRight:before {
  content: "\e687";
}

.icon-ArrowLineTopRight:before {
  content: "\e688";
}

.icon-ArrowLeftBold:before {
  content: "\e689";
}

.icon-UserAvatar:before {
  content: "\e68a";
}

.icon-ArrowCaretBottom:before {
  content: "\e68b";
}

.icon-ArrowLineBottomLeft:before {
  content: "\e68c";
}

.icon-PictureFilled:before {
  content: "\e68d";
}

.icon-RefreshRight:before {
  content: "\e68e";
}

.icon-ArrowSort2:before {
  content: "\e68f";
}

.icon-Check:before {
  content: "\e691";
}

.icon-ArrowLeft:before {
  content: "\e692";
}

.icon-Menu:before {
  content: "\e693";
}

.icon-ModeLight:before {
  content: "\e694";
}

.icon-CloseBold:before {
  content: "\e695";
}

.icon-ArrowLineBottom:before {
  content: "\e696";
}

.icon-BellMute:before {
  content: "\e697";
}

.icon-TipWarning:before {
  content: "\e698";
}

.icon-BellFill:before {
  content: "\e699";
}

.icon-ArrowDown:before {
  content: "\e69a";
}

.icon-TipCheck:before {
  content: "\e69b";
}

.icon-TipClose:before {
  content: "\e69c";
}

.icon-ModeDark:before {
  content: "\e69d";
}

.icon-ArrowCaretTop:before {
  content: "\e69e";
}

.icon-TipCloseFill:before {
  content: "\e69f";
}

.icon-Email:before {
  content: "\e6a0";
}

.icon-ArrowUpBold:before {
  content: "\e6a1";
}

.icon-ArrowRight:before {
  content: "\e6a2";
}

.icon-Close:before {
  content: "\e6a3";
}

.icon-Lock:before {
  content: "\e6a4";
}

.icon-Frame:before {
  content: "\e6a5";
}

.icon-EditPen:before {
  content: "\e6a6";
}

.icon-ArrowLineLeft:before {
  content: "\e6a7";
}

.icon-ChatRound:before {
  content: "\e6a8";
}

.icon-ArrowLineBottomRight:before {
  content: "\e6a9";
}

.icon-Setting:before {
  content: "\e6aa";
}

.icon-ThePlus:before {
  content: "\e6ab";
}

.icon-FullScreen:before {
  content: "\e6ac";
}

.icon-ArrowDownBold:before {
  content: "\e6ad";
}

.icon-Male:before {
  content: "\e6ae";
}

.icon-ArrowLineDownload:before {
  content: "\e6af";
}

.icon-ArrowSortTop:before {
  content: "\e6b0";
}

.icon-ChatDotRound:before {
  content: "\e6b1";
}

.icon-Picture:before {
  content: "\e6b2";
}

.icon-Unlock:before {
  content: "\e6b3";
}

.icon-ArrowUp:before {
  content: "\e6b4";
}

.icon-ArrowDoubleLeft:before {
  content: "\e6b5";
}

.icon-Calendar:before {
  content: "\e6b6";
}

.icon-ArrowRightBold:before {
  content: "\e6b7";
}

.icon-Refresh:before {
  content: "\e6b8";
}
