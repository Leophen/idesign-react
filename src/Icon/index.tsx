import React from 'react';
import './index.scss';
import classNames from 'classnames';

interface IconProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 组件自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 图标名称
   */
  name?: string;
  /**
   * 图标尺寸
   * @default 16
   */
  size?: number | string;
  /**
   * 图标颜色
   */
  color?: string;
  /**
   * 是否禁用按钮
   */
  disabled?: boolean;
  /**
   * 点击图标触发事件
   */
  onClick?: any;
}

const Icon: React.FC<IconProps> = (props) => {
  const {
    className,
    style,
    name = 'UserAvatar',
    size,
    color,
    disabled = false,
    onClick,
    ...restProps
  } = props;

  const iconStyles = {
    color,
    fontSize: size && parseInt(size.toString()),
    ...style,
  };

  return (
    <i
      style={iconStyles}
      className={classNames('i-icon', `icon-${name}`, disabled && 'i-icon-is-disabled', className)}
      onClick={
        !disabled
          ? onClick
          : () => {
              return;
            }
      }
      {...restProps}
    />
  );
};

export default Icon;
