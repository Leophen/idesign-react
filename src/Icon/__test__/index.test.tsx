import { render } from '@testing-library/react';
import Icon from '../index';

describe('Icon 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Icon name="ModeLight" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('size', () => {
    const { container } = render(<Icon name="TipCheck" size={16} />);
    expect(container.firstChild).toHaveStyle('font-size: 16px');
  })

  it('color', () => {
    const { container } = render(<Icon name="TipClose" color="#5e62ea" />);
    expect(container.firstChild).toHaveStyle('color: #5e62ea');
  })
});

