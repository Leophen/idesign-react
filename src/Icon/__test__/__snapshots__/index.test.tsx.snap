// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 组件测试 create 1`] = `
<DocumentFragment>
  <i
    class="i-icon icon-ModeLight"
  />
</DocumentFragment>
`;
